ivy/ivy_solver.py

```diff
@@ -583,11 +583,11 @@ def clauses_to_z3(clauses):
 
 def formula_to_z3_int(fmla):
 #    print "formula_to_z3_int: {} : {}".format(fmla,type(fmla))
-    if isinstance(fmla,ivy_logic.Definition or ivy_logic.is_eq(fmla) or isinstance(fmla,ivy_logic.Iff)):
```

Comment:
What does this code even do? It's passing an `or` to `isinstance`, but that `or` should always trivially be `ivy_logic.Definition`. Should the close paren be earlier?

Reply:
Yes, good catch. This is always `ivy_logic.Definition`, so this portion of the code only handles `Definition`s.

Equalities are, strangely enough, handled by the `ivy_logic.is_atom(fmla)` branch at the beginning of the function. The definition of `is_atom(term)` is `return (isinstance(term,App) or isinstance(term,Symbol)) and term.sort == lg.Boolean or isinstance(term,lg.Eq)`.

I'll fix this to actually behave properly.